src/OrchardCore/OrchardCore.ContentManagement/DefaultContentManager.cs

```diff
@@ -115,8 +115,8 @@ public async Task<IEnumerable<ContentItem>> GetAsync(IEnumerable<string> content
             {
                 contentItems[i] = await LoadAsync(contentItems[i]);
             }
-            
-            return contentItems;
+
+            return contentItems.OrderBy(c => contentItemIds.ToList().IndexOf(c.ContentItemId));
```

Comment:
I tested it via the `content_item_id` liquid filter which calls that method. Easy to repro:

1. Setup new site with blog recipe
2. Grab a couple of content item ids and note the associated document id
3. Add a liquid page with the following, listing the content item ids in an order other than by document id ascending:
```
{% assign itemIds = "4bfjky93de72k7qtpezeennwxv,405rvh88sz0972f788yjckce5c" | split: ',' %}
{% assign items = itemIds | content_item_id %}
{% for item in items %}
<div>
	{{ item.ContentItem.ContentItemId }}
</div>
{% endfor %}
```
4. The results are listed in order by document id rather than the order specified.

Maybe it can be fixed in yessql too?